code/src/reducers/tasks.js

```diff
@@ -0,0 +1,60 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+export const tasks = createSlice({
+    name: 'tasks',
+    initialState: {
+        items: []
+    },
+
+    //***** here is the Actions *****/
+    reducers: {
+        //state = current state
+        addTask: (state, action) => {
+
+            //id: Date:Now() gives it a millisecond based id
+            state.items.push({ id: Date.now(), text: action.payload })
+        },
+
+        toggleNeedsMore: (state, action) => {
```

Comment:
Maybe call this toggleCompleted or something? needsMore doesn't really fit to this project?